app/models/job_application.rb

```diff
@@ -169,17 +169,20 @@ def all_available_file_types
     @all_available_file_types ||= JobApplicationFileType.all.to_a
   end
 
+  # Return two arrays
+  # First with JobApplicationFile already existing
+  # Second with JobApplicationFileType
   def files_to_be_provided
     JobApplicationFileType.all.each_with_object([[], []]) do |file_type, memo|
-      existing = job_application_files.detect do |x|
-        x.job_application_file_type_id == file_type.id
-      end
+      existing_file = job_application_files.find_by(job_application_file_type: file_type)
```

Comment:
will that make an SQL query for each call? That's what I wanted to avoid in the first place :)

Reply:
That's right ! I will rollback this change sorry